src/ethernetModule.cpp

```diff
@@ -2,58 +2,49 @@
 
 EthernetUDP Udp;
 
+//GLOBALS
+byte macAddressTeensy[] = { 0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED };
+IPAddress ipAddressTeensy(10, 0, 0, 1);
+//IPAddress switchAddress(129, 241, 187, 1);
+IPAddress targetAddress(10, 0, 0, 2);
 
-namespace eth {
-
+unsigned int localPort = 8888;
+unsigned int targetPort = 8888;
 
-    byte macAddressTeensy[] = { 0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED };
-    IPAddress ipAddressTeensy(10, 0, 0, 1);
-    //IPAddress switchAddress(129, 241, 187, 1);
-    IPAddress targetAddress(10, 0, 0, 2);
+char emptyString[] = "";
+char recievedString[UDP_TX_PACKET_MAX_SIZE];
 
-    unsigned int localPort = 8888;
-    unsigned int targetPort = 8888;
-
-    char stringSent[] = "No packets recieved";
-    //char recieveConfirmed[] = "Packet recieved";
-    char recievedString[UDP_TX_PACKET_MAX_SIZE];
+namespace eth {
 
-    status ethernetSetup() { 
+    status setup() { 
         Ethernet.begin(macAddressTeensy,ipAddressTeensy);
-        int udpStatus = Udp.begin(localPort);
-        if (udpStatus == 1) {
-            return success;
+        if (Udp.begin(localPort)) {
+            return SUCCESS;
         }
         else {
-            return failure;
+            return FAILURE;
         }
     }
 
-    status ethernetCheck() {
-        int packetSize = Udp.parsePacket();
-        if (packetSize != 0) {                           //if part to confirm recieved UDP package
-            int udpStatus1 = Udp.read(recievedString, UDP_TX_PACKET_MAX_SIZE);
-            int udpStatus2 = Udp.beginPacket(targetAddress, targetPort);
-            Udp.write("Recieved UDP packet containing: ");
-            Udp.write(recievedString);
-            int udpStatus3 = Udp.endPacket();
-            if (udpStatus1 && udpStatus2 && udpStatus3) {
-                return success;
-            }
-            else {
-                return failure;
-            }
+    status write(char* message) {
+        int udpStatus1 = Udp.beginPacket(targetAddress, targetPort);
+        Udp.write(message);
+        int udpStatus2 = Udp.endPacket();
+        if (udpStatus1 && udpStatus2) {
+            return SUCCESS;
+        }
+        else {
+            return FAILURE;
+        }   
+    }
+    
+    char* read() {
+        if (Udp.parsePacket() != 0) {                           //if part to confirm recieved UDP package
+            Udp.read(recievedString, UDP_TX_PACKET_MAX_SIZE);
+            return recievedString;
         }
         else {
-            int udpStatus1 = Udp.beginPacket(targetAddress, targetPort);
-            Udp.write(stringSent);
-            int udpStatus2 = Udp.endPacket();
-            if (udpStatus1 && udpStatus2) {
-                return success;
-            }
-            else {
-                return failure;
-            }
+            return emptyString;
```

Comment:
Wouldn't 
```suggestion
            return nullptr;
```
 be a valid option here? I didn't think of this yesterday but at least doing a sanity check of the value after the function call should be easier.

Especially as you're using `char*` as a return type anyways.

Then you can also remove the global `emptyString` variable.